homeassistant/components/habitica/sensor.py

```diff
@@ -270,6 +302,94 @@ async def async_setup_entry(
         HabitipyTaskSensor(coordinator, description)
         for description in TASK_SENSOR_DESCRIPTION
     )
+    inventory_items_added: set[str] = set()
+
+    @callback
+    def add_entities() -> None:
+        """Add sensors for inventory items."""
+
+        nonlocal inventory_items_added
+
+        inventory = []
+        for key in coordinator.data.user["items"]["eggs"]:
+            if key in inventory_items_added:
+                continue
```

Comment:
Would it make sense to just create `set`s of those identifiers and directly compare those? I think that is a bit cleaner than to iterate over every egg and potion every coordinator update